### PR TITLE
Add config + validation to ORM and PRM

### PR DESCRIPTION
**Pull request by elrensmin**
## Summary

Brings ORM and PRM up to the config-driven, validation-logging standard already in the Preference RM (PR #476).

## Changes

- ORM: val_ratio/eval_interval in configs/orm.yaml; new evaluate_orm() logging val/loss + val/accuracy every eval_interval steps and at epoch end; train metrics namespaced train/*.
- PRM: new configs/prm.yaml; train_prm() refactored to take a Config; --config now required (breaking: CLI overrides removed); new evaluate_prm() logging val/loss + val/step_accuracy.
- Docs: updated reward_models/README.md and code/README.md.
- PRM samples now limits raw problems before step chunking, not packed records.

## Breaking change
train_prm CLI overrides removed. move them into a YAML config.

### CODE REVIEW

File: code/reward_models/train_prm.py [hunk 1]
```diff
@@ -49,20 +51,37 @@
...
-DEFAULT_LR = 5e-5
-DEFAULT_WARMUP_RATIO = 0.1
 DEFAULT_SEED = 13
 
 STEP_SEPARATOR = "\n<step>\n"
 PRM_CLASS_VALUES = [-1, 0, 1]  # Bad, Neutral, Good
 PRM_CLASS_TO_IDX = {value: idx for idx, value in enumerate(PRM_CLASS_VALUES)}
 
 
+def _stream_prm_jsonl(dataset_name: str, split: str):
```
Comment (elrensmin) [inline code/reward_models/train_prm.py:63]: datasets/pyarrow versions fail to parse PRM800K's schema, so we stream the raw JSONL directly from the Hub to keep the config-driven split working

**Review by natolambert — COMMENTED** [review]
*This review was written by Claude Code on Nathan's behalf.*

Thanks for this — it's in good shape overall. It faithfully mirrors the preference RM pattern from #476, the split-before-pack approach correctly prevents prompt/problem leakage between train and val in both scripts, and the docs/CHANGELOG are updated. I checked that `tasksource/PRM800K` really does store its data as `phase{1,2}_{train,test}.jsonl`, so the `HfFileSystem` workaround targets real paths (and the schema-cast issue it works around is real).

A few things before merging:

**1. Undisclosed hyperparameter changes.** The PR body doesn't mention that `orm.yaml` drops `lr` from 1e-4 to 5e-5, or that the PRM defaults changed substantially in the move from CLI defaults to `prm.yaml`: `epochs` 1→2, `max_steps` 20→12, `max_tokens` 5500→2048, `batch_size` 1→2 with `grad_accum` 16→8 (effective batch stays 16). The `max_tokens` cut especially changes what the model trains on — long traces get chunked much more aggressively. Did you run these configs? If you have loss/val curves, please share them; if the values are guesses, I'd rather keep the old ORM lr and PRM `max_tokens`/`epochs` here and land hyperparameter tuning as a separate PR.

**2. `samples` semantics quietly changed for PRM.** Before, the limit capped packed chunk records; now it caps raw problems before chunking, so `samples: 2000` produces more (and a variable number of) training examples than the same value used to. That's arguably the better definition, but please state it in the PR body and CHANGELOG entry since it changes what a given config does.

**3. Dead `try/except` in `demo_scoring` (train_prm.py).** `_stream_prm_jsonl` is a generator function, so calling it never raises — the first `try/except` block around `test_stream = _stream_prm_jsonl(...)` can't fire. Only the one around the iteration loop does anything; fold them into one.

**4. Silent failure path in `_stream_prm_jsonl`.** It swallows `FileNotFoundError` for both phase files, so a typo'd `dataset_name` (which the config now exposes) surfaces later as the generic "No PRM examples loaded" error. Raise if *neither* file was found, and/or note in the docstring that this loader only works with PRM800K-layout repos.

**Comment by elrensmin** [comment]
Sharing ORM runs at the two learning rates (1e-4 in purple). 

Screenshot from 2026-08-15 09-02-42

5e-5 gives a gentler backbone update, which is the safer default for a reward model whose signal will feed downstream RL; the slower initial loss drop lines up with that.

I reverted most of the PRM defaults except max_token to make it feasible on a single consumer GPU (max_token at 5500 on 32 GB VRAM I get OOMs which felt wrong for an educational baseline). Happy to revert ORM to lr=1e-4 and PRM to the original max_tokens defaults if you'd prefer this PR stay strictly structural. 

The rest of the suggestions are updated.

**Comment by natolambert** [comment]
Generally the ORM and PRM sections are pretty WIP and experimental, so I'm happy to have validated settings if the metrics look better. Let me look a little closer.
